src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,236 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class Parser {
+    protected String[] words;
+    private Ui ui;
+
+    public Parser(String command) {
+        this.words = command.split(" ");
+        ui = new Ui();
+    }
+
+    /**
+     * Analyses user's commands and performs the appropriate actions.
+     *
+     * @param tasks TaskList object consisting of user's existing tasks.
+     * @return String generated from the UI class. Aids in generation of DialogBox.
+     * @throws Exception if the command is Bye
+     */
+    public String validateParser(TaskList tasks) throws Exception {
```

Comment:
This method is quite long and violates the 30 LoC guideline ([source](https://nus-cs2103-ay2324s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods))

Perhaps you could consider moving the actions into separate methods of their own?

e.g:
```
if (Objects.equals(action, "todo")) {
   tryCreateTask();
}
```